File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */

var kthSmallest = function(root, k) {
    let path = []

    let inorder = (root, arr) => {
        if(root == null) return 

        if(root.left) inorder(root.left, arr)
        arr.push(root.val)
        if(root.right) inorder(root.right, arr)
    }
    
    inorder(root, path)

    console.log(path)
    return path[k - 1]
};

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        inorder = []

        self.dfs(root, inorder)

        return inorder[k - 1]

    
    def dfs(self, root, inorder):
        if root is None:
            return None
        
        self.dfs(root.left, inorder)
        inorder.append(root.val)
        self.dfs(root.right, inorder)
        